kube-client/src/lib.rs

```diff
@@ -470,6 +469,66 @@ mod test {
         Ok(())
     }
 
+    #[tokio::test]
+    #[ignore] // requires a cluster
+    async fn can_get_pod_metadata() -> Result<(), Box<dyn std::error::Error>> {
+        use crate::{
+            api::{DeleteParams, EvictParams, ListParams, Patch, PatchParams, WatchEvent},
+            core::subresource::LogParams,
+        };
+        use kube_core::{ObjectList, ObjectMeta};
+
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::default_namespaced(client);
+
+        // create busybox pod that's alive for at most 30s
+        let p: Pod = serde_json::from_value(json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": {
+                "name": "busybox-kube-meta",
+                "labels": { "app": "kube-rs-test" },
+            },
+            "spec": {
+                "terminationGracePeriodSeconds": 1,
+                "restartPolicy": "Never",
+                "containers": [{
+                  "name": "busybox",
+                  "image": "busybox:1.34.1",
+                  "command": ["sh", "-c", "sleep 30s"],
+                }],
+            }
+        }))?;
+
+        match pods.create(&Default::default(), &p).await {
+            Ok(o) => assert_eq!(p.name_unchecked(), o.name_unchecked()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+            Err(e) => return Err(e.into()),                         // any other case if a failure
+        }
+
+        let pod_metadata: kube_core::ObjectMeta = pods.get_metadata("busybox-kube-meta").await?.into();
+        assert_eq!(
+            "busybox-kube-meta",
+            pod_metadata
+                .name
+                .expect("Expected metadata to have a 'name' field")
+        );
+        assert_eq!(
+            Some((&"app".to_string(), &"kube-rs-test".to_string())),
+            pod_metadata
+                .labels
+                .expect("Expected metadata to have a 'name' field")
+                .get_key_value("app")
+        );
```

Comment:
test is great. maybe also do a patch-metadata in this fn as well!
watch stuff we can relegate to testing in runtime.

Reply:
Yeah, great call! For completeness I added a `list_metadata` test too. I was a bit apprehensive at first because there may be lots of pods running at the same time in the namespace. I managed to make it look nice through the `find` combinator.

Reply:
It should be fine to do a list 👍 . Integration tests are usually set up in a minimal environment on CI with just a few pods, and we warn in most places that integration tests hits your current context and may mutate.